### PR TITLE
Bug 1305

### PR DESCRIPTION
**Pull request by GuKKDevel**
might be on testserver, because Dirk, Ales and me  2017 tried to establish the new roots onto the testserver,
thereby we noticed, that server was gone

### CODE REVIEW

File: pages/index/3.php [hunk 1]
```diff
@@ -18,6 +18,37 @@
...
+	<li><?=_("SHA256 Hash:") ?> 412c5fa846da64a80148f788b5bb0b70517d6f12bfb133ae6a87cc6bd1921b90</li>
+</ul>
+
+<h3><?=_("Class 1 PKI Key")?></h3>
+<ul class="no_indent">
+	<li><a href="certs/root_256.crt"><?=_("Root Certificate (PEM Format)")?></a></li>
+	<li><a href="certs/root_256.der"><?=_("Root Certificate (DER Format)")?></a></li>
+	<li><a href="certs/root_256.txt"><?=_("Root Certificate (Text Format)")?></a></li>
+	<li><a href="<?=$_SERVER['HTTPS']?"https":"http"?>://crl.cacert.org/revoke.crl">CRL</a></li>
+	<li><?=_("SHA256 fingerprint:")?> 07ED BD82 4A49 88CF EF42 15DA 20D4 8C2B 41D7 1529 D7C9 00F5 7092 6F27 7CC2 30C5</li>
```
Comment (Golffies) [inline pages/index/3.php:37]: `<li><?=_("SHA1 fingerprint:") ?> DDFC DA54 1E75 77AD DCA8 7E88 27A9 8A50 6032 52A5</li>`

File: www/certs/class3_256.txt [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+         38:67:8b:90:fb:4d:0b:83:dc:48:f5:81:55:f5:2d:8c:6d:26:
+         a7:94:d5:25:bd:b0:78:52:f1:e4:7a:5d:29:e9:b1:ad:02:6a:
+         75:74:90:52:91:93:85:9b:46:7a:7a:4f:86:ef:0e:d1:d5:a4:
+         e2:7e:31:89:ad:dc:34:df:63:be:54:82:b0:0a:0b:bc:0d:db:
+         24:47:4c:34:07:af:32:75:99:f4:01:39:cc:9e:be:44:c6:f7:
+         16:91:90:6d:0a:04:1a:d8:db:d2:2a:b7:10:9e:56:aa:a3:d8:
+         9c:10:5e:17:7a:f2:3f:55:37:b3:95:bd:4b:8d:83:16:1d:57:
+         79:47:a0:b6:a7:8c:13:c9:50:48:33:c8:63:ac:b7:0a:88:28:
+         45:e3:71:91:26:d9:de:ef
+-----BEGIN CERTIFICATE-----
```
Comment (Golffies) [inline www/certs/class3_256.txt:101]: Discrepancy in www/certs/class3_256.txt:

The certificate in TEXT format is Class 3 Root SHA256 owning serial number 14
The certificate in PEM format attached here is Class 3 Root SHA256 owning serial number 0A:41:8A

Signatures of both certificates are correct although different.

**Review by Golffies — COMMENTED** [review]
In regards to "Bug-1305; Second part introducing the new cerificates":

The patch which you propose is based on Class 3 Root SHA256 owning serial number 14.
However, the Wiki makes available the same Class 3 Root SHA256 certificate, but owning another serial number: 0A:41:8A.

Signature of both certificates are different; both signatures are valid.

Wouldn't it be better to publish on the main CACert page the very same Class 3 Root certificate than the one already made available on the Wiki, i.e. the one with 0A:41:8A serial number ?

There is another discrepancy in www/certs/class3_256.txt file:

class3_256.txt encloses both:

- in text format, the certificate Class 3 Root SHA256 owning serial number 14;
- in PEM format, the certificate Class 3 Root SHA256 owning serial number 0A:41:8A.

Both certificates are present inside the same one file class3_256.txt

**Comment by GuKKDevel** [comment]
Thank you for that information I corrected this. new pull request https://github.com/CAcertOrg/cacert-devel/pull/11

**Pull request merged**